cmd/podman/libpodruntime/runtime.go

```diff
@@ -1,17 +1,59 @@
 package libpodruntime
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
 	"github.com/containers/storage"
 	"github.com/projectatomic/libpod/libpod"
 	"github.com/urfave/cli"
 )
 
 // GetRuntime generates a new libpod runtime configured by command line options
 func GetRuntime(c *cli.Context) (*libpod.Runtime, error) {
-	storageOpts := storage.DefaultStoreOptions
+	storageOpts, err := GetDefaultStoreOptions()
+	if err != nil {
+		return nil, err
+	}
 	return GetRuntimeWithStorageOpts(c, &storageOpts)
 }
 
+func GetRootlessStorageOpts() (storage.StoreOptions, error) {
+	var opts storage.StoreOptions
+
+	home := os.Getenv("HOME")
+	runtimePath := os.Getenv("XDG_RUNTIME_DIR")
+	if runtimePath == "" {
+		runtimePath = filepath.Join(home, "rundir")
+	}
+	if home == "" {
+		return opts, fmt.Errorf("HOME not specified")
+	}
+
+	opts.RunRoot = filepath.Join(runtimePath, "run")
+	dataDir := os.Getenv("XDG_DATA_DIR")
+	if dataDir != "" {
+		opts.GraphRoot = filepath.Join(dataDir, "containers", "storage")
+	} else {
+		opts.GraphRoot = filepath.Join(home, ".containers", "storage")
```

Comment:
If my argument against using `HOME` when calculating `XDG_RUNTIME_DIR` makes sense, this fallback here is the only portion of this function that would need the `HOME` value.  If we go that route, I'd like to move the `Getenv("HOME")` call and empty check into this block.